server/connectors/api-connector-mysql/src/main/scala/com/prisma/api/connector/mysql/database/DatabaseMutationBuilder.scala

```diff
@@ -106,13 +106,15 @@ object DatabaseMutationBuilder {
              create: DBIOAction[Any, NoStream, Effect],
              update: DBIOAction[Any, NoStream, Effect]) = {
 
-    val q = (sql"select exists" ++ DatabaseMutationBuilder.pathQueryForLastChild(projectId, updatePath)).as[Boolean]
+    val query = sql"select exists ( SELECT `id` FROM `#$projectId`.`#${updatePath.lastModel.name}` WHERE `id` = " ++ pathQueryForLastChild(projectId,
```

Comment:
pathQueryForLastChild Can either return a sub query or the id. This means that sometimes this code will generate an extra sub query. We need to clean this up later.

Reply:
Yeah, the next pass over the path logic should look into when we can simplify paths by looking for ids and when they are fully necessary to check relations. I tried to separate the checks from the paths, but once everything is cleaned up we can make sure we generate less complex queries wherever possible